Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,332 @@
+ * @param {Boolean} [options.createColorAttachments=true] Whether the FramebufferManager will construct its own color attachments.
+ * @param {Boolean} [options.createDepthAttachments=true] Whether the FramebufferManager will construct its own depth attachments.*
+ *
+ * @exception {DeveloperError} Cannot have both a depth and depth-stencil attachment.
+ *
+ * @private
+ * @constructor
+ */
+function FramebufferManager(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  this._colorAttachmentsLength = defaultValue(
+    options.colorAttachmentsLength,
+    1
+  );
+
+  this._color = defaultValue(options.color, true);
+  this._depth = defaultValue(options.depth, false);
+  this._depthStencil = defaultValue(options.depthStencil, false);
+  this._supportsDepthTexture = defaultValue(
+    options.supportsDepthTexture,
+    false
+  );
+  //>>includeStart('debug', pragmas.debug);
+  if (!this._color && !this._depth && !this._depthStencil) {
+    throw new DeveloperError(
+      "must enable at least one type of framebuffer attachment."
+    );
+  }
+  if (this._depth && this._depthStencil) {
+    throw new DeveloperError(
+      "Cannot have both a depth and depth-stencil attachment."
+    );
+  }
+  //>>includeEnd('debug');
+
+  this._createColorAttachments = defaultValue(
+    options.createColorAttachments,
+    true
+  );
+  this._createDepthAttachments = defaultValue(
+    options.createDepthAttachments,
+    true
+  );
+
+  this._width = undefined;
+  this._height = undefined;
+  this._pixelDatatype = undefined;
+
+  this._framebuffer = undefined;
+  this._colorTextures = undefined;
+  if (this._color) {
+    this._colorTextures = new Array(this._colorAttachmentsLength);
+  }
+  this._depthStencilRenderbuffer = undefined;
+  this._depthStencilTexture = undefined;
+  this._depthRenderbuffer = undefined;
+  this._depthTexture = undefined;
+}
+
+Object.defineProperties(FramebufferManager.prototype, {
+  framebuffer: {
+    get: function () {
+      return this._framebuffer;
+    },
+  },
+  status: {
+    get: function () {
+      return this._framebuffer.status;
+    },
+  },
+});
+
+FramebufferManager.prototype.isDirty = function (width, height, pixelDatatype) {
+  var dimensionChanged = this._width !== width || this._height !== height;
+  var attachmentsNeedUpdate =
+    (this._color && !defined(this._colorTextures[0])) ||
+    (this._color && !this._createColorAttachments) ||
+    (this._depth && !this._createDepthAttachments) ||
+    (this._depthStencil && !this._createDepthAttachments);
+  var datatypeChanged =
+    defined(pixelDatatype) && this._pixelDatatype !== pixelDatatype;
+
+  return dimensionChanged || attachmentsNeedUpdate || datatypeChanged;
+};
+
+FramebufferManager.prototype.update = function (
+  context,
+  width,
+  height,
+  pixelDatatype,
+  pixelFormat
```

Comment:
It's only used in `PostProcessStageTextureCache`, where I believe a framebuffer's pixel datatype and format can never change at runtime (otherwise a new framebuffer would be created in `getFramebuffer()`...). I've added both to constructor options, which should simplify that code a bit.